File: Foundation/week_02-week_04/JavaBasicsProject/src/week_03/OO/SharpieSet/SharpieSet.java
package week_03.OO.SharpieSet;

import java.util.ArrayList;
import java.util.List;

public class SharpieSet {
  //  Reuse your Sharpie class
  //  Create SharpieSet class
  //  it contains a list of Sharpie
  //  it has a method that adds a new Sharpie to the set: add(sharpie)
  //  it has a method that returns the number of usable Sharpies: countUsable() -> sharpie is usable if it contains ink
  //  it has a method that removes all unusable Sharpies: removeTrash()

  List<Sharpie> sharpieList = new ArrayList<>();

  public void add(Sharpie sharpie) {
    sharpieList.add(sharpie);
  }

  public int countUsable() {
    int usable = 0;
    for (Sharpie sharpie : sharpieList) {
      if (sharpie.inkAmount != 0) {
        usable++;
      }
    }
    return usable;
  }

  public void removeTrash() {
    for (int i = 0; i < sharpieList.size(); i++) {
      if (sharpieList.get(i).inkAmount == 0) {
        sharpieList.remove(i);
        i--;
      }
    }
  }

}

File: Foundation/week_02-week_04/PythonBasicsProject/week_03/oop/sharpie_set/sharpie_set.py
# - Create `SharpieSet` class
#   - it contains a list of Sharpie
#   - count_usable() -> sharpie is usable if it has ink in it
#   - remove_trash() -> removes all unusable sharpies

class SharpieSet():
    def __init__(self, sharpie_list=[]):
        self.sharpie_list = sharpie_list

    def count_usable(self):
        counter = 0
        for sharpie in self.sharpie_list:
            if sharpie.ink_amount > 0:
                counter += 1
        return counter

    def remove_trash(self):
        i = 0
        while i < len(self.sharpie_list):
            if self.sharpie_list[i].ink_amount < 0:
                self.sharpie_list.pop(i)
                i -= 1
            i += 1

